src/Mapping/Event/Adapter/ORM.php

```diff
@@ -95,6 +95,11 @@ public function getObjectManager()
             throw new \LogicException(sprintf('Event args must be set before calling "%s()".', __METHOD__));
         }
 
+        // @todo create getObjectManager in UploadableBaseEventArgs.php and remove this check
```

Comment:
Could you please use the same approach than #2547, creating and commenting the method, and adding the `@method` annotation at `UploadableBaseEventArgs`?
https://github.com/doctrine-extensions/DoctrineExtensions/blob/c8a995317bb0a3de88f64f912c8d2ca7b76095e4/src/Tree/Node.php#L12-L54

Reply:
@phansys sure I could do that, but at that point, what's the point in even keeping the check in place?
Current master already restricts doctrine/orm to 2.14+, when the method is implemented in UploadableBaseEventArgs, it will always exist.

Reply:
I think the better way to go about it is to create the method, change the pipeline to ignore getObjectManager calls on EventArgs (it currently ignores getEventManager calls) and get rid of the check. It doesn't actually hurt backward compatibility in any way, or am I missing something?

Reply:
It sounds logic at first sight.
Let me check out the PR locally to have a better understanding of the scenario here, as I'm not following this topic.

Reply:
From what I see, the case seems to be even worse than I thought, as the property used here (`ORM::$args`) is using a type (`Doctrine\Common\EventArgs`) that does not guarantee the `getEntityManager()` or the `getEntityManager()` methods:
https://github.com/doctrine-extensions/DoctrineExtensions/blob/c8a995317bb0a3de88f64f912c8d2ca7b76095e4/src/Mapping/Event/Adapter/ORM.php#L52

AFAIK, the only usable class declaring these methods is [`Doctrine\Persistence\Event\LifecycleEventArgs`](
https://github.com/doctrine/persistence/blob/8a710c21dfe6e88df21c371012be1bf6f2337525/src/Persistence/Event/LifecycleEventArgs.php#L16), as the [implementation for the ORM](https://github.com/doctrine/orm/blob/9f555ea8fb4f7ae9a362285dc71e85776a1c5cc3/lib/Doctrine/ORM/Event/LifecycleEventArgs.php#L19) is deprecated.

That said, I know we surely couldn't use `LifecycleEventArgs` as type declaration here, as there are more implementations of `EventArgs` passed here (`Doctrine\ORM\Event\LoadClassMetadataEventArgs` by instance).

With that on mind, I guess we could use some type check and a deprecation path on these getters to prevent calling `$this->args->getObjectManager()` and `$this->args->getObject()` on objects with other type than `Doctrine\Persistence\Event\LifecycleEventArgs`.

I mean something like this:
```php
public function getObject(): object
{
    if (null === $this->args) {
        throw new \LogicException(sprintf('Event args must be set before calling "%s()".', __METHOD__));
    }

    if (!$this->args instanceof \Doctrine\Persistence\Event\LifecycleEventArgs) {
        @trigger_error(sprintf(
            'Calling "%s()" on an event that is not of type "%s" is deprecated since gedmo/doctrine-extensions 3.x'
            .' and will throw a "%s" error in version 4.0.',
            __METHOD__,
            LifecycleEventArgs::class,
            \TypeError::class
        ), E_USER_DEPRECATED);
    }

    // @todo Remove this check when `\Doctrine\Persistence\Event\LifecycleEventArgs` is used.
    if (\method_exists($this->args, 'getObject')) {
        return $this->args->getObject();
    }

    return $this->args->getEntity();
}

Please, confirm if my understanding about the current situation is right and let me know if I'm missing something.
Thank you.

Reply:
> From what I see, the case seems to be even worse than I thought, as the property used here (ORM::$args) is using a type (Doctrine\Common\EventArgs) that does not guarantee the getEntityManager() or the getEntityManager() methods

Correct, that's why these warnings are ignored for class in the php actions here.

> That said, I know we surely couldn't use LifecycleEventArgs as type declaration here, as there are more implementations of EventArgs passed here (Doctrine\ORM\Event\LoadClassMetadataEventArgs by instance).

I cannot confirm that is the case, but if it is, the current implementation would crash anyway because those EventArgs do not have a `getEntity` method.

From my understanding (and I could definitely be wrong there) all EventArgs actually passed from Doctrine to EventListeners are of type Persistence/LifecycleEventArgs. The only way I could see this breaking bc is if people implemented their own EventArgs implementations with getEntityManager and getEntity methods and these were somehow passed to our ORM Adapter.

Reply:
I'll just add some deprecation notices if the $args don't have a getObjectManager/getObject method and see if these are ever triggered to confirm. Give me a few days to test it.

Reply:
Thank you very much for the time and effort put on this!

In the test I used, I can confirm other types than `Doctrine\Persistence\Event\LifecycleEventArgs` are passed to this class, like `Doctrine\ORM\Event\LoadClassMetadataEventArgs`.

Reply:
Time to figure out why that doesn't cause a crash then. `getEntity` must just never be called on those.

Reply:
Edit: actually, being on my pc, there's a very simple explanation for why that would still be okay:
https://github.com/doctrine/orm/blob/2.14.x/lib/Doctrine/ORM/Event/LoadClassMetadataEventArgs.php